### DISCUSSION TITLE
Parse string prop as markdown

### DISCUSSION THREAD

**Discussion by edward1127** [mdx-js/mdx#2274] [Q&A] [answered]
How can I parse a string prop (props.content in the example below) as markdown within a mdx file? 

```
import React, { useState } from "react";
import Dropdown from "/components/Dropdowns/Dropdown";

# My MDX page

**This is an unordered list**

- Item One
- Item Two
- Item Three

{props.content}


export default ({ children }) => <div className="prose">{children}</div>;
```

**Comment by remcohaszing** [comment, marked as answer]
MDX does not compile props. If you pass props as a string, you get them as a string. If you want to treat the content as markdown, you can use something like [`react-markdown`](https://github.com/remarkjs/react-markdown).

**Reply by edward1127** [reply]
thanks for the reply.

**Reply by edward1127** [reply]
I'm trying to parse some custom syntaxs such as [[number-input]] and replace them with react input components. In that case, I think I might need to find some other packages?

**Reply by remcohaszing** [reply]
You can do this with `react-markdown` and a remark plugin. However, it’s generally not recommended to invent your own syntax. Have you considered using [`rehype-raw`](https://github.com/rehypejs/rehype-raw) or [`remark-directive`](https://github.com/remarkjs/remark-directive)?

**Reply by edward1127** [reply]
thanks for the suggestions! I will check those packages.

**Comment by wooorm** [comment]
You shouldn’t, probably.

It’s the same question: “Parse string prop as JavaScript”. You shouldn’t. You can write JavaScript in JavaScript. You shouldn’t use strings for that. You can do it of course, with an eval function (we have [one too](https://mdxjs.com/packages/mdx/#evaluatefile-options)), but it’s typically unsafe, slow, and a really bad idea.

**Reply by wooorm** [reply]
Practically, you define `props.content` somewhere.
*That’s* where you shouldn’t make it a string, but React elements or whatever.
That’s the place you need to fix.

**Reply by edward1127** [reply]
Got it! Thanks!

**Reply by donny-bras** [reply]
@wooorm I didn't quite get what you meant. I need to pass a markdown string to a component, how can I do it?
Could you please, give an example? Thanks!

**Reply by wooorm** [reply]
No you probably don’t. If you are, you are more likely to be doing it wrong. Don’t. If you really have to, read the docs: `evaluate`.

Use MDX.

Also, I have no clue what you are doing. So I have no clue how to help you.
